demos/console.php

```diff
@@ -8,28 +8,80 @@ class Test extends \atk4\data\Model
 
     public function generateReport()
     {
-        $this->log('info', 'Starting long process');
-        $this->debug('test=123');
-        sleep(1);
-        $this->debug('test=321');
-        sleep(5);
+        $this->log('info', 'Console will automatically pick up output from all DebugTrait objects');
+        $this->debug('debug {foo}', ['foo'=>'bar']);
+        $this->emergency('emergency {foo}', ['foo'=>'bar']);
+        $this->alert('alert {foo}', ['foo'=>'bar']);
+        $this->critical('critical {foo}', ['foo'=>'bar']);
+        $this->error('error {foo}', ['foo'=>'bar']);
+        $this->warning('warning {foo}', ['foo'=>'bar']);
+        $this->notice('notice {foo}', ['foo'=>'bar']);
+        $this->info('info {foo}', ['foo'=>'bar']);
 
         return 123;
     }
 }
 
-$app->add('Console')->set(function ($console) {
+$tt = $app->add('Tabs');
+
+$t = $tt->addTab('set()');
+$t->add([
+    'Header',
+    'icon'=> 'terminal',
+    'Console output streaming',
+    'subHeader'=> 'any output your PHP script produces through console is displayed to user in real-time',
+]);
+$t->add('Console')->set(function ($console) {
     $console->output('Executing test process...');
     sleep(1);
     $console->output('Now trying something dangerous..');
     sleep(1);
+    echo 'direct output is captured';
+
+    throw new \atk4\data\Exception('BOOM - exceptions are caught');
+});
 
-    throw new \atk4\data\Exception('BOOM');
-    $console->output('hello there');
+$t = $tt->addTab('runMethod()', function ($t) {
+    $t->add([
+        'Header',
+        'icon'=> 'terminal',
+        'Non-interractive method invocation',
+        'subHeader'=> 'console can invoke a method, which normaly would be non-interractive and can still capture debug output',
+    ]);
+    $t->add('Console')->runMethod($t->add(new Test()), 'generateReport');
```

Comment:
```
atk4\core\Exception: Method is not defined for this object
              class: "Test"
             method: "emergency"
          arguments: ["emergency {foo}",{"foo":"bar"}]
Stack Trace: 
dor\atk4\core\src\DynamicMethodTrait.php:  35 - atk4\core\Exception atk4\core\Exception::__construct
                    ("Method is not defined for this object")
\wamp64\www\git\atk\ui\demos\console.php:  13 - Test atk4\data\Model::__call()
                                        :     - Test Test::generateReport()
```

Method `emergency` is not defined ?!

Reply:
oh it needs https://github.com/atk4/core/pull/77

Reply:
https://github.com/atk4/core/pull/77 merged in